src/main/java/com/quaz/ticket/screeningseat/ScreeningSeat.java

```diff
@@ -0,0 +1,32 @@
+package com.quaz.ticket.screeningseat;
+
+import com.quaz.ticket.persistence.AbstractEntity;
+import com.quaz.ticket.reservation.Reservation;
+import com.quaz.ticket.screening.Screening;
+import com.quaz.ticket.screeningroom.Seat;
+import jakarta.persistence.Entity;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@Entity
+@Table(name = "screening_seats")
+public class ScreeningSeat extends AbstractEntity<Long> {
+
+    @ManyToOne(optional = false)
+    private Screening screening;
+
+    @ManyToOne(optional = false)
+    private Seat seat;
+
+    @ManyToOne
+    private Reservation reservation;
+
+    public boolean isAvailable() {
+        return reservation == null;
```

Comment:
Jak rozumiem ten model zakłada, że dodając nowy screening, trzeba wrzucić do bazy tyle rekordów ile jest miejsc, nawet jeśli nikt ich jeszcze nie zarezerwował? Czy nie lepiej byłoby założyć, że brak wpisu oznacza brak rezerwacji i dopiero po rezerwacji robić wpis?

Reply:
Zastanawiałem się nad takim model, ale uznałem, że to może być tyci łatwiejsze, kosztem narzutu na zajmowanie miejsca w bazie danych. To co bym dopisał to generowanie tych miejsc do rezerwacji przez jakąś scheduled metodę, to samo tyczy się archiwizacji (lub usuwania) miejsc z seansów, które już minęły. Taki sposób tworzenia dostępnych miejsc może być wygodny jeśli chcielibyśmy dodawać logikę do dostępnych miejsc, dla przykładu, w czasie blisko po Covidowym moglibyśmy dodać politykę w metodzie generacji miejsc, że dostępne są co drugie miejsce albo, że całe rzędy są zablokowane. Zamiast na stałe zapisywać taką logikę do fizycznych miejsc.